### ISSUE TITLE
Mark Success/Failure from the UI uses wrong settings

### ISSUE THREAD

**Issue by KevinKobi** [apache/airflow#8431]
**What happened**:
In the UI the default behavior is Downstream and Recursive set for Clear:
![Screen Shot 2020-04-17 at 18 41 39](https://user-images.githubusercontent.com/63675983/79587565-65109480-80db-11ea-9d27-d9ff32a5d5a9.png)

these preset choices should effect only clear button but they effect all the others.
By choosing `Mark success` or `Mark Failed` it will take the choice of `Downstream` and `Recursive` from `Clear` unless removing them explicitly. 

**What you expected to happen**:

Each row has it's own settings. As the UI shows Mark Success has it's own Downstream Upstream buttons. It should use it's own not other rows settings.

As alternative, at the end the window perform only ONE action (Clear, Mark Success, Mark Failure) the block of 4 * 3  = 12 buttons is not needed:
![Screen Shot 2020-04-17 at 18 47 26](https://user-images.githubusercontent.com/63675983/79588057-1283a800-80dc-11ea-89a5-0841e0493caa.png)
It can have just 4 buttons for all the three actions (`Clear`, `Mark Success`, `Mark Failed`) and Clear can have extra setting of `Recursive` (I'm not sure why `Recursive` available only in the Clear row)

I'm not UX expert so there might be better way to do this but it feels that there is redundancy with the buttons (which also create this bug)

**How to reproduce it**:

1. Add any DAG
2. in the UI click on any task to open the menu
3. Choose `Mark success` or `Mark Failure` it will show you list of all tasks met with `Recursive` and `Downsteam` though you did not choose these settings.
4. Go back, remove the preset of `Recursive` and `Downsteam`  from `Clear` row. Click on `Mark success` or `Mark Failure` now it will show you only the task you choose.

**Comment by vikramkoka** [comment]
@ryanahamilton Can you please comment on this?

**Comment by ryanahamilton** [comment]
@KevinKobi I believe I resolved the issue in the refactoring I did (#10944) prior to 2.0 release (screenshot of the updated presentation below). Instead of utilizing a single form for all of the options + actions, each row of options + action is a single form so it will no longer be POSTing values from sibling rows/forms. Your additional comments on improving the experience by removing redundancy are still valid even with this fixed. 

![image](https://user-images.githubusercontent.com/3267/106792678-6e47ff00-6624-11eb-9ac7-151088ef883c.png)
